packages/api/src/external/carequality/patient.ts

```diff
@@ -1,48 +1,86 @@
+import {
+  getPatientDiscoveryResultCount,
+  getPatientDiscoveryResults,
+} from "../../command/medical/patient-discovery-result/get-patient-discovery-result";
+import { CQLink, PatientCQDataCreate } from "../../domain/medical/cq-patient-data";
 import { Organization } from "../../domain/medical/organization";
-import { Patient, PatientExternalData } from "../../domain/medical/patient";
+import { Patient } from "../../domain/medical/patient";
+import { PatientDiscoveryResult } from "../../domain/medical/patient-discovery-result";
+import { Product } from "../../domain/product";
+import { EventTypes, analytics } from "../../shared/analytics";
 import { capture } from "../../shared/notifications";
-import { makeIheGatewayAPI } from "./api";
 import { toFHIR } from "../fhir/patient";
-import { patientToIheGateway } from "./patient-conversion";
-import { setCarequalityId } from "./patient-external-data";
-import { PatientDataCarequality } from "./patient-shared";
-import { MedicalDataSource } from "..";
-
-const createContext = "cq.patient.create";
-
-export function getCQData(
-  data: PatientExternalData | undefined
-): PatientDataCarequality | undefined {
-  if (!data) return undefined;
-  return data[MedicalDataSource.CAREQUALITY] as PatientDataCarequality;
-}
+import { cqOrgsToXCPDGateways } from "./organization-conversion";
+import { createPatientDiscoveryRequest } from "./create-pd-request";
+import { makeIheGatewayAPI } from "./api";
+
+dayjs.extend(duration);
+
+const createContext = "cq.patient.discover";
+export const PATIENT_DISCOVERY_TIMEOUT = dayjs.duration({ minutes: 2 });
+const CHECK_DB_INTERVAL = dayjs.duration({ seconds: 5 });
+type RaceControl = { isRaceInProgress: boolean };
 
 export async function discover(
   patient: Patient,
   organization: Organization,
   facilityNPI: string
 ): Promise<void> {
-  let iheGateway: IHEGateway | undefined;
   try {
-    console.log(`IHE Gateway patient discovery - M patientId ${patient.id}`);
-
-    iheGateway = makeIheGatewayAPI();
+    const { id, cxId } = patient;
+    const iheGateway = makeIheGatewayAPI();
+    console.log(`Kicking off patient discovery for patientId: ${id}`);
 
     const fhirPatient = toFHIR(patient);
 
-    const iheGatewayRequest = patientToIheGateway({
+    const nearbyCQOrgs = await searchNearbyCQOrganizations({
+      cxId,
+      patientId: id,
+    });
+    const xcpdGateways = cqOrgsToXCPDGateways(nearbyCQOrgs);
+
+    const pdRequest = createPatientDiscoveryRequest({
       patient: fhirPatient,
       cxId: patient.cxId,
-      xcpdGateways: [], // TODO ADD XCPD GATEWAYS WHEN WE HAVE THEM
-      facilityNPI: facilityNPI,
+      xcpdGateways,
+      facilityNPI,
       orgName: organization.data.name,
       orgOid: organization.oid,
     });
 
-    iheGateway.startPatientDiscovery(iheGatewayRequest);
+    // Intentionally asynchronous - we will be checking for the results in the database
+    iheGateway.startPatientDiscovery([pdRequest]);
```

Comment:
Whats going to be the release plan for this? Because AFAIK we wont be able to merge this into prod until the infra is setup